values.go

```diff
@@ -189,8 +189,11 @@ func (v Bytes) String() string {
 //
 type Character string
 
-func NewCharacter(b string) Character {
-	return Character(b)
+func NewCharacter(b string) (Character, error) {
+	if !format.IsValidCharacter(b) {
+		return "_", fmt.Errorf("invalid character: %s", b)
```

Comment:
Best to follow the Go idiom of returning the empty value for the type, i.e., for `string`, and empty `""`. Callers should only use one of the return values (first if second/error is nil, or only the second/error if non-nil), similar to https://github.com/onflow/cadence/pull/1392/files#diff-e9b60dd44adee3235a03e234d28b9711ab20c09f79ea62bc6be1d924dc59a4ecR258

```suggestion
		return "", fmt.Errorf("invalid character: %s", b)
```

If really needed, maybe consider returning [REPLACEMENT CHARACTER / U+FFFD](https://en.wikipedia.org/wiki/Specials_(Unicode_block)#Replacement_character) instead of the arbitrary underscore `_`